components/post_view/post_list/post_list.jsx

```diff
@@ -147,6 +147,7 @@ export default class PostList extends React.PureComponent {
     }
 
     componentWillUnmount() {
+        this.markChannelAsReadAndViewed(this.props.channelId);
```

Comment:
I'm trying to think if there's anything else that could cause this component to unmount that wouldn't represent the user having interacted with the app, but I can't think of anything that wouldn't already happen with the current logic.